lib/util.js

```diff
@@ -779,14 +764,15 @@ function formatCollectionIterator(ctx, value, recurseTimes, visibleKeys, keys) {
 }
 
 function formatPromise(ctx, value, recurseTimes, visibleKeys, keys) {
-  var output = [];
-  var internals = inspectPromise(value);
-  if (internals.status === 'pending') {
+  const output = [];
+  const [state, result] = binding.getPromiseDetails(value);
+
+  if (state === binding.kPending) {
     output.push('<pending>');
   } else {
     var nextRecurseTimes = recurseTimes === null ? null : recurseTimes - 1;
-    var str = formatValue(ctx, internals.value, nextRecurseTimes);
-    if (internals.status === 'rejected') {
+    var str = formatValue(ctx, result, nextRecurseTimes);
+    if (state === binding.kRejected) {
```

Comment:
Also fyi @Fishrock123, this seems like a cleaner way to inspect Promise states than what the abort-on-rejected PR currently does…